text/0000-compiler-requirements.md

```diff
@@ -2,18 +2,45 @@
 
 ## Current State
 
+The entire font compilation pipeline is written in Python, with some parts being written in Cython for speed.
+
 ### Current Tools
 
+fontmake. It's the glue that binds together ufo2ft, fontTools and some other codebases to take any of the usual font formats (UFOs, Designspace + UFOs and Glyphs.app files) and produce font binaries (TTF, CFF(2); static, interpolated static, variable).
+
 ### Key Limitations
 
+* Python is bad at embarrasingly parallel tasks like instance generation and glyph compilation due to overhead and lacking multicore facilities.
+* The memory consumption of the compilation process can hinder the amount of build processes running in parallel.
+* It's easy to run into performance problems with OpenType Layout table repacking, which quickly consumes a large part of the compile time.
+* Codebases have grown organically by adding new things on top, some parts are byzantine and hard to change, making optimizations or rearranging code paths hard.
+    * VF generation requires compiling all masters fully and then merging them, instead of generating variable data directly.
+    * fontmake is underloved from a maintenance perspective, with some low hanging performance fruits continuing to hang.
```

Comment:
I'm not sure this is true. The last time we had this discussion (https://github.com/fonttools/fonttools/issues/1095) we concluded there weren't many low-hanging fruit.